src/DotLiquid/Util/StrFTime.cs

```diff
@@ -16,79 +16,116 @@ public static class StrFTime
                                              + "(?<" + GROUP_WIDTH + ">[1-9][0-9]*)?"
                                              + "(?<" + GROUP_DIRECTIVE + @">[a-zA-Z%])";
 
+        private delegate string DateObjectDelegate(object dateTime, CultureInfo culture);
         private delegate string DateTimeDelegate(DateTime dateTime, CultureInfo culture);
-        private delegate string DateTimeOffsetDelegate(DateTimeOffset dateTimeOffset, CultureInfo culture);
-
-        private static readonly Dictionary<string, DateTimeDelegate> Formats = new Dictionary<string, DateTimeDelegate>
+        
+        private static readonly Dictionary<string, DateObjectDelegate> Formats = new Dictionary<string, DateObjectDelegate>
         {
-            { "a", (dateTime, culture) => dateTime.ToString("ddd", culture) },
-            { "A", (dateTime, culture) => dateTime.ToString("dddd", culture) },
-            { "b", (dateTime, culture) => dateTime.ToString("MMM", culture) },
-            { "B", (dateTime, culture) => dateTime.ToString("MMMM", culture) },
-            { "c", (dateTime, culture) => dateTime.ToString("ddd MMM dd HH:mm:ss yyyy", culture) },
-            { "C", (dateTime, culture) => ((int)Math.Floor(Convert.ToDouble(dateTime.ToString("yyyy", culture))/100)).ToString(culture) },
-            { "d", (dateTime, culture) => dateTime.ToString("dd", culture) },
-            { "D", (dateTime, culture) => dateTime.ToString("MM/dd/yy", culture) },
-            { "e", (dateTime, culture) => dateTime.ToString("%d", culture).PadLeft(2, ' ') },
+            { "a", (dateTime, culture) => string.Format(culture, "{0:ddd}", dateTime) },

```

Comment:
Is it me or is there no matching test for "a" format? As we are changing the line and the format from `ddd` to `{0:ddd}` I was wondering if the output remains indeed the same (should be).

Looks like reformatting to adhere to a code style, removing ambiguity of extension methods and avoiding any possible null deref.

Reply:
Reformatting is to allow for various date types to utilize the same codebase rather than have a version for each primitive type. Test is one of the test cases listed on TestToStrFTime_DateFormat, specifically, `[TestCase("%a", ExpectedResult = "Sun")] // Abbreviated weekday name (Mon)`